[DOCS] Update shared attributes for Elasticsearch

lcawl opened the pull request:
This pull request removes shared attributes from the Versions.asciidoc file and instead adds a reference to the list of shared attributes in the elastic/docs/shared repository.

lcawl commented:
@debadair If we eventually want to use the shared attributes from books other than the Elasticsearch Reference (e.g. from the Painless and client books), we can move the "include::{docs-dir}/shared/attributes.asciidoc[]" from index-shared1.asciidoc to Versions.asciidoc.  I haven't done any work on those other books at this point.

debadair commented:
BTW, I'd be inclined to pull the attributes for the definitive guide and painless and plugins docs into the shared file, as well.

lcawl commented:
@debadair verified that the other books were okay to use the shared attributes, so I moved the reference to the elastic/docs/shared/attributes.asciidoc from index-shared1.asciidoc to Versions.asciidoc.  I also moved the definitive guide, painless, and plugins doc URLs to the shared file too.

debadair commented:
LGTM

lcawl commented:
Thanks!  I've retrofitted to 5.x and 5.5
